modules/instrument_builder/js/instrument_builder.instrument.js

```diff
@@ -120,11 +120,20 @@ var Instrument = {
 
                         // Add dropdown and special naming when no date format is set
                         // (i.e when addDateElement() is used)
-                        if (element.Options.dateFormat === "") {
+                        if (!element.Options.dateFormat) {
+                          if (elName === '' || !elName.includes('_date')) {
```

Comment:
should elName ever be an empty string?  its also strange im seeing that the standard date option appends '_date' to the name you provide, but not for the other two options (basic and month)

Reply:
@zaliqarosli the element name is used as the column name in the database table for the instrument. So it cannot be empty is my belief and thus it shouldn't be.

Reply:
thats what i thought. so how come you're checking for if its empty?

Reply:
@zaliqarosli might not be needed to check if empty. I thought for sanity reasons in case it ever happens to be. It's similar to me usually checking if something is undefined first.

Reply:
i think it's okay to check if its empty, but i don't think the step after should be to just append '_date' and not provide a nonempty $elName

Reply:
@zaliqarosli yah I think the logic is messed up. I think the standard is to have {something}_date. I'm going to look why I added this again and maybe make sure to put a comment if there is a reason.